src/config/lockfile.py

```diff
@@ -74,3 +106,113 @@ def write(self):
         """Writes the state of this lockfile object to the file."""
         with open(self._path, 'w', encoding='utf-8') as f:
             f.write(json.dumps(self, indent=2))
+
+    def add_entry(self, entry: LockfileEntry, entry_type: Asset):
+        """Adds a specified entry to the lockfile object."""
+        if self.get_entry(entry, entry_type) is None:
+            if type is Asset.MOD:
+                self.lockfile_entries.mods[entry.name] = entry
+            elif type is Asset.RESOURCE:
+                self.lockfile_entries.resoucepacks[entry.name] = entry
+            elif type is Asset.TEXTURE:
+                self.lockfile_entries.texturepacks[entry.name] = entry
+            else:
+                self.lockfile_entries.shaderpacks[entry.name] = entry
+        else:
+            raise ClickException(f'{entry_type} {entry.name} ' +
+                                 'already exists in lockfile')
+
+    def remove_entry(self, entry: LockfileEntry, entry_type: Asset):
+        """Removes a given entry from the lockfile object.
+
+        Returns the removed entry, or None if entry was not found.
+        """
+        if self.get_entry(entry, entry_type) is not None:
+            if type is Asset.MOD:
+                removed = self.lockfile_entries.mods.pop(entry.name)
+            elif type is Asset.RESOURCE:
+                removed = self.lockfile_entries.resoucepacks.pop(entry.name)
+            elif type is Asset.TEXTURE:
+                removed = self.lockfile_entries.texturepacks.pop(entry.name)
+            else:
```

Comment:
```
elif type is Asset.SHADERPACK:
  ...
else:
  raise Exception('Unknown asset type encountered!')
```
here and below. Raise this exception and don't catch it so it crashes to indicate something is wrong with your code.